fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/api/dimension/EntityPlacer.java

```diff
@@ -0,0 +1,32 @@
+package net.fabricmc.fabric.api.dimension;
+
+import net.minecraft.block.pattern.BlockPattern;
+import net.minecraft.entity.Entity;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.Direction;
+
+/**
+ * Responsible for placing an Entity once they have entered a dimension.
+ * Stored by a FabricDimensionType, and used in Entity::changeDimension.
+ *
+ * @see FabricDimensions
+ * @see FabricDimensionType
+ */
+@FunctionalInterface
+public interface EntityPlacer {
+
+    /**
+     * Handles the placement of an entity going to a dimension.
+     *
+     * <p> This method may have side effects such as the creation of a portal in the target dimension,
+     * or the creation of a chunk loading ticket.
+     *
+     * @param portalDir the direction the portal is facing, may be {@code null} if no portal was used
+     * @param portalX   the x coordinate of the corner of the portal, 0 if no portal was used
+     * @param portalY   the y coordinate of the corner of the portal, 0 if no portal was used
+     * @return a teleportation target, or {@code null} to fall back to further handling
+     * @apiNote When this method is called, the entity's world is its source dimension.
+     */
+    /* @Nullable */
+    BlockPattern.TeleportTarget placeEntity(Entity teleported, ServerWorld destination, Direction portalDir, double portalX, double portalY);
```

Comment:
Minecraft uses Y as the vertical coordinate in case you were thinking it was the horizontal one. At any rate, Z should be included as well since modders might want to know the z-coordinate as well.

Reply:
I will be honest, I am not entirely sure what those values are. They come from `Entity#lastPortalDirectionVector`, and are used by portals to determine where to spawn the entity. After a closer look, `portalX` can be either an `x` coordinate or a `z` coordinate, depending on the portal direction... Not sure what to do with those honestly.

Reply:
Update: I am dumb ! Those are the horizontal and vertical offsets of the player relative to the top front left corner of the portal.
```suggestion
    BlockPattern.TeleportTarget placeEntity(Entity teleported, ServerWorld destination, Direction portalDir, double horizontalOffset, double verticalOffset);
```